lib/spack/spack/compilers/msvc.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+import re
+import subprocess
+import sys
+from distutils.version import StrictVersion
```

Comment:
@johnwparent / @scheibelp why did this ever pass review? It was deprecated in Python 3.10. We have our own version utilities. Now Spack is forward incompatible with Python 3.12.

Reply:
> Rather inconvenient this PR made Spack depend on distutils...

This is an utterly dramatic phrasing. This PR did not entrench a distutils dependency irreparably in the depths of Spack: it is it is easily removed as an import (in the absolute worst case, that specific logic could be vendored in, but there must be many ways to handle it - as you say, we have our own version utilities).

> why did this ever pass review?

I fully contest both your wording and the sentiment behind it: that we must be so hyper-vigilant that we analyze line by line all possible future problems that would be caused. That is completely untenable as an approach to developing software.

Finally, this PR was merged approximately 1.5 years ago: stop dropping random comments in dead PRs and raise an issue (feel free to reference the PR in the issue).